### ISSUE TITLE
Test case 4 (last customer stuck in queue)

### ISSUE THREAD

**Issue by eesean** [nus-cs2030/2223-s2#237]
Hi all,

Would like to ask whether anyone has come across a similar issue as shown in the pic? My last customer does not seem to want to leave the queue.

<img width="314" alt="stuck in queue" src="https://user-images.githubusercontent.com/110977266/220915145-b9516798-eb1b-49ec-8530-a0ff2cd122b0.png">

I guess what I find peculiar is why my code works for every customer except for the last customer

**Comment by trwstin** [comment]
How and where did you write the code that updates the serverList after a waiting customer is served?

**Comment by eesean** [comment]
For each server I have a priority queue of customers, when I poll the queue and the customer number corresponds to the head event (e.g. Event 1) of my priority queue of events, I use a method to transit to the next event (serve) and add the event back to the queue of events.
My Event Comparator compares time stamp followed by priority of events, hence when I poll again I should get Event 1 again, this time with the Serve event, during which I serve the waiting customer.

**Comment by Lofthesaver** [comment]
To debug, you can try to print out the PQ and see what are all the Events added to the PQ and whether there are some events missing (i.e., a WaitEvent isn't returning the nextEvent, etc.)

**Comment by eesean** [comment]
Yep I managed to fix my edge case! The event corresponding to the customer in waiting queue but not at head of the queue was removed during the "iteration" of event queue polling hence the event went missing.

**Comment by kelleyhan25** [comment]
> Yep I managed to fix my edge case! The event corresponding to the customer in waiting queue but not at head of the queue was removed during the "iteration" of event queue polling hence the event went missing.

sorry to bother! was wondering how you managed to fix the removal of the event? bc/ i'm running into the same problem..

**Comment by eesean** [comment]
Hi so sorry I did not see your query yesterday! What happened was when I polled the Wait Event corresponding to customer 6, I did not factor a case whereby the server is not available (my next event method back then is wait --> serve), hence that event was not added back to the priority queue after polling. What's important is to be able to handle a wait event under this scenario such that all customers (and waiting time) are accounted for, such as adding the event back to the queue. 

Hope this helps and all the best!
